### ISSUE TITLE
[2019] 

### ISSUE THREAD

**Issue by cfiken** [cfiken/paper-reading#36]
### タイトル
Frustratingly Easy Natural Question Answering
### 著者
Lin Pan, Rishav Chakravarti, Anthony Ferritto, Michael Glass, Alfio Gliozzo, Salim Roukos, Radu Florian, Avirup Sil
(IBM)
### 投稿日付
2019/09/11
## 1. どんなもの？

Natural Questions といった難しい QA タスクにおいて、Leaderboard 上位にいるものの論文で説明されていないような手法が用いているテクニックなどを紹介した論文。
Attention-over-Attention, Data Augmentation, Ensemble strategy, pretrain 方法など、スコアを上げるための手法を説明する。
Natural Questions で SoTA を更新。

## 2. 先行研究と比べてどこがすごい？

Attention-over-Attention を始めとした、論文なのでは触れられていない QA のスコアを改善するためのテクニックを紹介しつつ、実験で検証した。

## 3. 技術や手法のキモはどこ？

- Attention-over-Attention
    - question の tokens Q, context の tokens C に対して、M = CQ^T を計算
    - M を row-wise, column-wise にそれぞれ softmax を取り、β, α とする
    - βを row で平均をとり、 s = α^T β ^T を計算する
    - 上記の手続きを BERT の output に対して 2heads で行い、得られた s_1, s_2 を start, end の logits 計算に重み付き和の形で加える
- Data Augmentation
    - 先行研究である SOS (Sentence Order Shuffling)
    - 既存データセット (SQuAD 2.0, NewsQA, TriviaQA) のランダム抽出と、NQに似たデータを抽出したもの
- Ensemble 戦略
    - Seed アンサンブルと複数モデルのアンサンブル それぞれ実験
- hyper parameters tuning で stride や sampling rate を微調整
    - stride は 192 に
    - sampling rate は回答可能かどうかで変えた

## 4. どうやって有効だと検証した？

Natural Questions で実験し、先行研究の SoTA を上回った

## 5. 議論はある？

- Learning rate や Batch size は少し変えただけで大きく結果が異なるため、今回はちゃんと調べてないとのこと
- BERT 以外の pre-train モデルを試したい

## 6. 次に読むべき論文は？

- Synthetic QA Corpora Generation with Roundtrip Consistency
    - https://arxiv.org/abs/1906.05416
    - Chris Alberti, Daniel Andor, Emily Pitler, Jacob Devlin, Michael Collins (Google Research)
    - ACL2019
    - BERT joint を Data Augmentation でスコア改善した研究
